### PR TITLE
Prepare checkout-ui-extensions package release

### PR DESCRIPTION
**Pull request by simontaisne**
### Background

This release brings the latest `checkout-ui-extension` packages from `checkout-web`. The main focus of this update is to improve the checkout UI extensions API docs.

### Solution

**Features**

- New `Checkout::PaymentMethod::RenderModal` extension point (private preview)
- New `useAvailablePaymentOptions()` and `useSelectedPaymentOptions()` payment options hooks

**Docs**

- Bump `@shopify/generate-docs` version
- Improve various API descriptions and examples
- Refactor React hooks docs by using `descriptionType`
- Update configurations content
- Cleanup the mentions to protected customer data
- Add example for multiple extension points

### 🎩

You can tophat the packages by following these [instructions](https://vault.shopify.io/page/Extension-Libraries~gkKh.md#user-content-2-tophat-the-packages).

To preview the docs that will be generated from this commit, you can tophat [the spin instance](https://shopify-dev.checkout-web-api-docs-mybn.simon-taisne.eu.spin.dev/docs/api/checkout-ui-extensions) 🌀

### Checklist

- [x] I have :tophat:'d these changes
- [x] I have updated relevant documentation

### CODE REVIEW

File: packages/checkout-ui-extensions/src/extension-points/extension-points.ts [hunk 1]
```diff
@@ -89,6 +89,15 @@ export interface ExtensionPoints {
...
   >;
+  /**
+   * A static extension point that renders a form modal when a buyer selects the custom onsite payment method.
+   *
+   * @private
+   */
+  'Checkout::PaymentMethod::RenderModal': RenderExtension<
+    StandardApi<'Checkout::PaymentMethod::RenderModal'>,
+    AllComponents
+  >;
```
Comment (simontaisne) [inline packages/checkout-ui-extensions/src/extension-points/extension-points.ts:100]: Is it fine to release this new `@private` extension point publicly?

**Comment by jamesvidler** [comment]
There is this [PR open right](https://github.com/Shopify/checkout-web/pull/21109/files) now which has some doc updates for Payment options. @kumar303 is this a blocker for this PR?

**Comment by jamesvidler** [comment]
@oliverigor are there any changes here to the components that need to be reviewed?

**Comment by simontaisne** [comment]
@kumar303 I included the most recent updates for Payment options 👍 Let me know if I can merge this one before publishing!

**Review by kumar303 — APPROVED** [review]
🚀

**Pull request merged**